File: easymapper/src/test/java/test/easymapper/AutoParameterizedTest.java
package test.easymapper;

import autoparams.AutoSource;
import autoparams.customization.Customization;
import autoparams.mockito.MockitoCustomizer;
import autoparams.processor.InstancePropertyWriter;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;
import org.junit.jupiter.params.ParameterizedTest;

@Target({ElementType.ANNOTATION_TYPE, ElementType.METHOD})
@Retention(RetentionPolicy.RUNTIME)
@ParameterizedTest
@AutoSource
@Customization({
    MockitoCustomizer.class,
    InstancePropertyWriter.class
})
public @interface AutoParameterizedTest {
}

File: easymapper-kotlin/src/test/kotlin/easymapper/kotlin/AutoParameterizedTest.kt
package easymapper.kotlin

import autoparams.kotlin.AutoKotlinSource
import org.junit.jupiter.params.ParameterizedTest

@Target(AnnotationTarget.FUNCTION)
@Retention(AnnotationRetention.RUNTIME)
@ParameterizedTest
@AutoKotlinSource
annotation class AutoParameterizedTest
